aim/web/api/runs/object_api_utils.py

```diff
@@ -86,6 +90,11 @@ def set_ranges(self, record_range: IndexRange, record_density: int,
         else:
             self.total_record_range = self._calculate_ranges()
 
+        if record_step is not None:
+            if record_step < 0:
+                record_step += self.total_record_range.stop
```

Comment:
do we have a real use-case for this? if so, still need to check that modulo of record_step is in the allowed range.

Reply:
the alternative to specifying -1 as a default would be `None` and checking that ether `record_range` or `record_step` is not None

Reply:
yeah that was the initial method, yet i think its confusing

Reply:
I added assertion just in case